### PR TITLE
Generate list of tags

### LINKED ISSUE #39: Generate the list of tags
**I'm submitting a:**  
- [ ] bug report
- [x] feature request

**Description**:

Create a list of tags that will help in developing several other features. To create the list of tags, you may use the `folder name` as a tag and dynamically, generate the list of tags.
Consider the following path: `code/search/src/linear_search`:For this path, we will get two tags `search` and `linear search`. 

Just like the `meta-data`, the list of tags is expected to be stored in a similar format for further use.

This will give us the ability to organize our search results in a better way.

### PR DESCRIPTION
**Pull request by ankan17**
**Checklist**

- [x] My branch is up-to-date with the upstream `develop` branch.
- [ ] I have added necessary documentation (if appropriate).

**Which issue does this PR fix?**: 

Fixes: #39

First steps for issues #40, #21 

**What this PR does?**:

A function has been added to update app which generates a list of
tags in the project from the metadata.json file. A new property
is defined in the settings file which stores the path to the tags
json file. A tags.json file has already been generated for the
existing metadata.json.

**TODOs (if any)**:

- Prefill search bar
- Provide real-time search suggestions

### CODE REVIEW

File: update/views.py [hunk 1]
```diff
@@ -13,6 +13,17 @@ def update_metadata(updated_at):
     with open(settings.METADATA_JSON, 'w') as f:
         json.dump(data, f)
 
+def update_tags(updated_at):
+    topics = []
+    data = {'updated_at' : updated_at}
+    for keys in json.load(open(settings.METADATA_JSON)):
+        for topic in keys.split('/'):
+            if topic not in (topics + ['unclassified', 'updated_at']):
```
Comment (AdiChat) [inline update/views.py]: You must ignore `test` and `src` as well.

File: update/views.py [hunk 2]
```diff
@@ -13,6 +13,17 @@ def update_metadata(updated_at):
...
+                topics.append(topic)
+    data['tags'] = list(map(lambda v: v.replace('_', ' ').lower(), topics))
```
Comment (AdiChat) [inline update/views.py]: You may, also, consider `-` along with `_` as some inconsistencies may exist.

See `graph-algorithm`

**Comment by AdiChat** [comment]
Great work 👍 

The issue is that `metadata.json` in this repository has not been updated for some time. So, you may update the metadata to re-generate the accurate list of tags. 
There have been some structural changes in Cosmos which is not reflected in the metadata such as support for folders `src` and `test`.

Kindly make the changes and it will be good to be merged 👍

File: update/views.py (hunk 1 above)
Comment (ankan17) [reply update/views.py]: `test` and `src` were not there in the `metadata.json` file and are not included in the `tags.json` file

File: update/views.py (hunk 2 above)
Comment (ankan17) [reply update/views.py]: There were 2-3 cases where '-' was there and in some cases I thought it was necessary for the '-' to be there. So should I convert all '-' to ' ' too? @AdiChat

**Comment by ankan17** [comment]
@AdiChat How can I update the `metadata.json` file? Isn't it updated from the `cosmos` directory? And I think that was added to `.gitignore` because I don't see any content in it. 😅

File: update/views.py (hunk 2 above)
Comment (AdiChat) [reply update/views.py]: Yes, convert all `-` to ` `. 👍

**Comment by AdiChat** [comment]
In this case, you need to clone Cosmos and run the concerned part of the script locally to generate the metadata.

**Comment by ankan17** [comment]
@AdiChat I have updated the `metadata.json` file and also excluded `src` and `test` while generating the `tags.json`. I think this completes all the changes required. If there is anymore, please review :)

**Comment by sakshee-19** [comment]
@ankan17  I just want to ask have you updated ```metadata.json``` manually?

**Comment by ankan17** [comment]
@sakshee-19 I cloned the cosmos repo and ran the update_metadata function manually like @AdiChat asked.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**